paparazzi/paparazzi/src/main/java/app/cash/paparazzi/internal/resources/RepositoryLoader.kt

```diff
@@ -0,0 +1,1175 @@
+                  }
+                }
+                resourceValue.addValue(valueName, numericValue, "")
+              }
+            }
+          }
+        }
+      }
+    )
+
+    resourceValue.setFormats(formats)
+    resourceValue.namespaceResolver = namespaceResolver
+    val visibility = getVisibility(ATTR, name)
+
+    // todo kzheng handle attrNamespace
+    return BasicResourceItem(type = ATTR, name, visibility, configuration, resourceValue)
+  }
+
+  @Throws(
+    IOException::class,
+    XmlPullParserException::class,
+    XmlSyntaxException::class
+  )
+  private fun createPluralsItem(
+    name: String,
+    file: PathString,
+    configuration: RepositoryConfiguration
+  ): BasicResourceItem {
+    val resourceValue = object : PluralsResourceValueImpl(namespace, name, null, null) {
+      // Allow the user to specify a specific quantity to use via tools:quantity
+      override fun getValue(): String {
+        val quantity = parser.getAttributeValue(
+          ATTR_QUANTITY,
+          TOOLS_URI
+        )
+        if (quantity != null) {
+          val value = getValue(quantity)
+          if (value != null) {
+            return value
+          }
+        }
+        return super.getValue()
+      }
+    }
+    val defaultQuantity = parser.getAttributeValue(TOOLS_URI, ATTR_QUANTITY)
+    val namespaceResolver = parser.namespaceResolver
+    forSubTags(
+      TAG_ITEM,
+      object : XmlTagVisitor {
+        override fun visitTag() {
+          val quantityValue = parser.getAttributeValue(null, ATTR_QUANTITY)
+          if (quantityValue != null) {
+            val quantity = Arity.getEnum(quantityValue)
+            if (quantity != null) {
+              val text = textExtractor.extractText(parser, false)
+              resourceValue.addPlural(quantityValue, text)
+            }
+          }
+        }
+      }
+    )
+    if (defaultQuantity != null) {
+      val defaultArity = Arity.getEnum(defaultQuantity)
+      if (defaultArity == null || resourceValue.getValue(defaultArity.name) == null) {
+        throw XmlSyntaxException(
+          "Invalid value of the " + namespaceResolver.prefixToUri(TOOLS_URI) + ':' + ATTR_QUANTITY + " attribute.",
+          parser,
+          getDisplayName(file)
+        )
+      }
+    }
+    val visibility = getVisibility(PLURALS, name)
+    resourceValue.namespaceResolver = namespaceResolver
+    return BasicResourceItem(type = PLURALS, name, visibility, configuration, resourceValue)
+  }
+
+  @Throws(IOException::class, XmlPullParserException::class)
+  private fun createStringItem(
+    type: ResourceType,
+    name: String,
+    configuration: RepositoryConfiguration,
+    withRowXml: Boolean
+  ): BasicResourceItem {
+    val namespaceResolver = parser.namespaceResolver
+    val text = if (type == ID) null else textExtractor.extractText(parser, withRowXml)
+    val rawXml = if (type == ID) null else textExtractor.getRawXml()
```

Comment:
```suggestion
    val text = if (type == ResourceType.ID) null else textExtractor.extractText(parser, withRowXml)
    val rawXml = if (type == ResourceType.ID) null else textExtractor.getRawXml()
```

fully qualified where possible, for readability.